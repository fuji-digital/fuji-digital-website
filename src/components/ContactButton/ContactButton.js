import { StyledContactButton, StyledArrowIcon } from "./ContactButton.styles";
import arrowIcon from "../../images/icons/arrow-icon.svg";

export const ContactButton = ({ text }) => {
  return (
    <StyledContactButton>
      <div>{text} </div>
      <StyledArrowIcon
        src={arrowIcon}
        alt="A black and white arrow"
      ></StyledArrowIcon>
    </StyledContactButton>
  );
};
