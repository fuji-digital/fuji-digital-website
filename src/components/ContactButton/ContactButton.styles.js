import styled from "styled-components";

export const StyledContactButton = styled.button`
  background-color: white;
  display: flex;
  align-items: center;
  justify-content: space-around;
  font-family: "Poppins", sans-serif;
  font-size: 1rem;
  border-radius: 50px;
  padding: 0.5rem 0.2rem 0.5rem 3rem;
  width: 13rem;
  height: 3rem;
`;

export const StyledArrowIcon = styled.img`
  height: 2rem;
`;
