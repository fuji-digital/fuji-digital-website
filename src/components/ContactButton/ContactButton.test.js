import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { ContactButton } from "./ContactButton";

describe("ContactButton", () => {
  it("matches snapshot", () => {
    const component = renderer
      .create(<ContactButton text="the button" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("text inside button is correct", () => {
    render(<ContactButton text="Contact Us" />);
    const textContactButton = screen.getByRole(`button`, {
      name: /Contact Us/i,
    });
    expect(textContactButton).toBeInTheDocument();
  });
});
