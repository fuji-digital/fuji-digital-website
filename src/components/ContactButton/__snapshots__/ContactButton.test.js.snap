// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactButton matches snapshot 1`] = `
<button
  className="sc-aYaIB koDWfZ"
  data-testid="contact-button"
  onClick={[Function]}
>
  <div>
    the button
  </div>
  <img
    alt="A black and white arrow"
    className="sc-gEvDqW dTdWeV"
    src="arrow-icon.svg"
  />
</button>
`;
